### ISSUE TITLE
Keyword 'super' inside an anonymous inner class causes an error

### ISSUE THREAD

**Issue by claeslundahl** [cincheo/jsweet#363]
To replicate the error: check out a new copy of https://github.com/cincheo/jsweet-quickstart.git. Then change the main method to this:
```
    public static void main(String[] args) throws NoSuchFieldException, IllegalAccessException {
        // you can use regular Java API
        List<String> l = new ArrayList() {
            final ArrayList me = this;

            @Override
            public boolean add(Object o) {
                return super.add(o);
            }
        };
        l.add("Hello");
        l.add("world");
    }
```
After that, as usual, "mvn generate-sources".
Then an error is cast. It includes the text 'error TS2335:" super "can only be referred to in a derived class.'

Kind regards
/C

**Comment by renaudpawlak** [comment]
That's right... this case is not supported yet with the default generation. The reason is that since by default there is no runtime, JSweet erases the JDK classes, so you cannot extend them. This could be improved however because it is possible to support some degree of inheritance of JDK classes even without a runtime (JSweet already supports some specific cases, but not with inner classes).

In short, at this point, you have several workarounds:

- Do not subclass JDK classes, use delegation instead (it is anyway good practice to use delegation rather than inheritance in many contexts, right?)... Note that you can also use delegation to wrap a JDK class in your own class, then use inheritance (anonymous classes) on your JDK wrapper.
- If you really want to subclass JDK classes directly (no delegation at all), you need to put the J4TS JDK runtime in your classpath. Then the ``ArrayList`` class will exist at runtime and you will be able to create subclasses of it.

I am keeping this as a potential enhancement for next milestones. Thanks!

**Comment by claeslundahl** [comment]
Thank you for your feedback. I will see if we can proceed with the suggestions you gave.

I'm trying to figure out if we can migrate our script from GWT to Jsweet instead. There are quite a few classes to go through and we have more errors that need to be resolved - unclear sometimes even what the error is or how it can be described.

Sincerely
/ C
